src/main/java/org/thymeleaf/util/NumberUtils.java

```diff
@@ -285,6 +285,71 @@ private static DecimalFormatSymbols computeDecimalFormatSymbols(
     
     
     
+    public static String formatCurrency(final Number target, final Locale locale) {
+
+        Validate.notNull(locale, "Locale cannot be null");
+
+        if (target == null) {
+            return null;
+        }
+
+        NumberFormat format = null;
+        
+        if (target instanceof Double || target instanceof Float || target instanceof BigDecimal ||
```

Comment:
Yeah, not sure why this `if` statement is like that in the original `formatNumber` either - `NumberFormat` can format any standard `Number` type.

Might be something to consider cleaning up afterwards, outside of this PR.